test/jdk/java/awt/print/PrinterJob/PrinterDevice.java

```diff
@@ -23,22 +23,34 @@
 
 /*
  *
- * @bug 4276227
+ * @test
+ * @bug 4276227 8320443
  * @summary Checks that the PrinterGraphics is for a Printer GraphicsDevice.
  * Test doesn't run unless there's a printer on the system.
- * @author prr
+ * @key printer
  * @run main/othervm PrinterDevice
  */
 
-import java.awt.*;
-import java.awt.geom.*;
-import java.awt.print.*;
-import java.io.*;
-import javax.print.attribute.*;
-import javax.print.attribute.standard.*;
+import java.io.File;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.GraphicsConfiguration;
+import java.awt.GraphicsDevice;
+import java.awt.Rectangle;
+import java.awt.geom.AffineTransform;
+import java.awt.print.PageFormat;
+import java.awt.print.Printable;
+import java.awt.print.PrinterException;
+import java.awt.print.PrinterJob;
+import javax.print.attribute.HashPrintRequestAttributeSet;
+import javax.print.attribute.PrintRequestAttributeSet;
+import javax.print.attribute.standard.Destination;
+import javax.print.attribute.standard.OrientationRequested;
 
 public class PrinterDevice implements Printable {
 
+    static volatile boolean failed = false;
```

Comment:
Is it really needed? In all the cases where you assign `true` to the `failed` field, you also explicitly throw an exception, which is enough to fail the test.

Reply:
It is to test the "the macOS printing implementation was swallowing exceptions it should not."

Reply:
Not sufficient if something catches the exception as used to happen